### PR TITLE
removal of unused variables/typedef

### PR DESCRIPTION
**Pull request by JoanThibault**
include/sycl/algorithm/
	- sort.hpp
		replace int by size_t in lines 87, 88
	- exclusive_scan.hpp
		replace int by size_t in lines 114, 115
	- inner_product.hpp
		remove lines 128, 129 : unused variables (localid, globalid)
	- tranform_reduce.hpp
		remove line 66 : unused type
		remove lines 85, 86 : unused variables (localid, globalid)
	- inclusive_scan.hpp
		replace int by size_t in lines 92, 93
	- count_if.hpp
		remove line 66 : unused type
		remove lines 86, 87 : unused variables (localid, globalid)
	- for_each.hpp
		remove line 53 : unused type

### CODE REVIEW

File: include/sycl/algorithm/exclusive_scan.hpp [hunk 1]
```diff
@@ -111,8 +111,8 @@ OutputIterator exclusive_scan(ExecutionPolicy &sep, InputIterator b,
       h.parallel_for<
           cl::sycl::helpers::NameGen<1, typename ExecutionPolicy::kernelName> >(
           r, [aI, aO, bop, vectorSize, i](cl::sycl::nd_item<3> id) {
-            int td = 1 << (i - 1);
-            int m_id = id.get_global(0);
+            size_t td = 1 << (i - 1);
```
Comment (AdamHarries) [inline include/sycl/algorithm/exclusive_scan.hpp]: Is there a particular reason that `size_t` is preferable here?

Comment (AdamHarries) [inline include/sycl/algorithm/exclusive_scan.hpp]: Aside from being the type that `id.get_global(0)` returns

**Comment by AdamHarries** [comment]
Builds and tests fine, and there don't seem to be any damaging changes made, so I'd reccomend accepting and merging.

**Comment by Ruyk** [comment]
Looks fine to me, size_t is the actual value returned by get_global on the SYCL specification.

**Pull request merged**